ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -86,41 +94,52 @@
    */
   void start() throws InterruptedException {
     long startTimeMs = time.milliseconds();
+    final AtomicInteger numFailures = new AtomicInteger(0);
     try {
-      File mountPath = new File(disk.getMountPath());
-      if (mountPath.exists()) {
-        List<Thread> startupThreads = new ArrayList<>();
-        final AtomicInteger numFailures = new AtomicInteger(0);
-        for (final Map.Entry<PartitionId, BlobStore> partitionAndStore : stores.entrySet()) {
-          Thread thread = Utils.newThread("store-startup-" + partitionAndStore.getKey(), new Runnable() {
-            @Override
-            public void run() {
-              try {
-                partitionAndStore.getValue().start();
-              } catch (Exception e) {
-                numFailures.incrementAndGet();
-                metrics.totalStoreStartFailures.inc();
-                logger.error("Exception while starting store for the partition" + partitionAndStore.getKey(), e);
-              }
-            }
-          }, false);
-          thread.start();
-          startupThreads.add(thread);
-        }
-        for (Thread startupThread : startupThreads) {
-          startupThread.join();
-        }
-        if (numFailures.get() > 0) {
-          logger.error(
-              "Could not start " + numFailures.get() + " out of " + stores.size() + " stores on the disk " + disk);
+      checkMountPathAccessible();
+
+      List<Thread> startupThreads = new ArrayList<>();
+      for (final Map.Entry<PartitionId, BlobStore> partitionAndStore : stores.entrySet()) {
+        Thread thread = Utils.newThread("store-startup-" + partitionAndStore.getKey(), () -> {
+          try {
+            partitionAndStore.getValue().start();
+          } catch (Exception e) {
+            numFailures.incrementAndGet();
+            logger.error("Exception while starting store for the partition" + partitionAndStore.getKey(), e);
+          }
+        }, false);
+        thread.start();
+        startupThreads.add(thread);
+      }
+      for (Thread startupThread : startupThreads) {
+        startupThread.join();
+      }
+      if (numFailures.get() > 0) {
+        logger.error(
+            "Could not start " + numFailures.get() + " out of " + stores.size() + " stores on the disk " + disk);
+      }
+
+      // DiskSpaceAllocator startup. This happens after BlobStore startup because it needs disk space requirements
+      // from each store.
+      List<DiskSpaceRequirements> requirementsList = new ArrayList<>();
+      for (BlobStore blobStore : stores.values()) {
```

Comment:
I added a test for successful startup with segmented logs and an induced diskspaceallocator failure case.